diff -Naur tunneldigger-9999-orig/broker/main.py tunneldigger-9999/broker/main.py
--- tunneldigger-9999-orig/broker/main.py	2016-10-15 14:41:26.000000000 +0200
+++ tunneldigger-9999/broker/main.py	2016-10-15 14:42:42.884134573 +0200
@@ -28,7 +28,7 @@
     'version': 1,
     'formatters': {
         'simple': {
-            'format': '[%(levelname)s/%(name)s] %(message)s',
+            'format': '%(asctime)s [%(levelname)s/%(name)s] %(message)s',
         },
     },
     'handlers': {
