diff -Naur alfred-announce-orig/announce.sh alfred-announce-9999/announce.sh
--- alfred-announce-orig/announce.sh	2016-04-18 12:08:11.000000000 +0200
+++ alfred-announce-9999/announce.sh	2016-04-18 12:00:58.000000000 +0200
@@ -1,9 +1,7 @@
 #!/bin/bash
 
-#set -x #echo on
-
 DIR="$(dirname "$0")"
-ALFRED=/usr/local/sbin/alfred
+ALFRED=/usr/sbin/alfred
 ANNOUNCE_OPTS=""
 
 (
@@ -15,20 +13,16 @@
   case $1 in
     -u)
       shift
-      #alfred-json -z -r 158 -s $1 > /dev/null
-      #test $? -ne 0 && exit
       ALFREDSOCKET="-u $1"
       ;;
     -h|--help)
-      echo "Usage: $0 [-i <ifname>] [-b <batadv-dev>] [-f <fastd-dev>] [-u <alfred-socket>] [-s <site_code>]"
+      echo "Usage: $0 [-u <alfred-socket>]"
       exit
       ;;
   esac
   shift
 done
 
-#./announce.py -d nodeinfo.d/   ${ANNOUNCE_OPTS}  | python -m json.tool
-#./announce.py -d statistics.d/ ${ANNOUNCE_OPTS}  | python -m json.tool
 ./announce.py -d nodeinfo.d/   ${ANNOUNCE_OPTS} | gzip | $ALFRED $ALFREDSOCKET -s 158
 ./announce.py -d statistics.d/ ${ANNOUNCE_OPTS} | gzip | $ALFRED $ALFREDSOCKET -s 159
 
