diff -Naur batman-adv-2016.3-orig/net/batman-adv/main.h batman-adv-2016.3/net/batman-adv/main.h
--- batman-adv-2016.3-orig/net/batman-adv/main.h	2016-09-01 16:14:46.000000000 +0200
+++ batman-adv-2016.3/net/batman-adv/main.h	2016-10-10 18:16:01.509774754 +0200
@@ -167,7 +167,7 @@
 /* Maximum number of fragments for one packet */
 #define BATADV_FRAG_MAX_FRAGMENTS 16
 /* Maxumim size of each fragment */
-#define BATADV_FRAG_MAX_FRAG_SIZE 1400
+#define BATADV_FRAG_MAX_FRAG_SIZE 1280
 /* Time to keep fragments while waiting for rest of the fragments */
 #define BATADV_FRAG_TIMEOUT 10000
 
