diff -Naur batman-adv-2017.0.1-orig/compat-include/linux/average.h batman-adv-2017.0.1/compat-include/linux/average.h
--- batman-adv-2017.0.1-orig/compat-include/linux/average.h	2017-03-07 16:38:09.000000000 +0100
+++ batman-adv-2017.0.1/compat-include/linux/average.h	2017-03-15 08:27:15.789161046 +0100
@@ -77,7 +77,7 @@
 	static inline void ewma_##name##_add(struct ewma_##name *e,	\
 					     unsigned long val)		\
 	{								\
-		unsigned long internal = ACCESS_ONCE(e->internal);	\
+		unsigned long internal = ACCESS_ONCE_RW(e->internal);	\
 		unsigned long weight_rcp = ilog2(_weight_rcp);		\
 		unsigned long precision = _precision;			\
 									\
@@ -86,7 +86,7 @@
 		BUILD_BUG_ON((_precision) > 30);			\
 		BUILD_BUG_ON_NOT_POWER_OF_2(_weight_rcp);		\
 									\
-		ACCESS_ONCE(e->internal) = internal ?			\
+		ACCESS_ONCE_RW(e->internal) = internal ?		\
 			(((internal << weight_rcp) - internal) +	\
 				(val << precision)) >> weight_rcp :	\
 			(val << precision);				\
